### PR TITLE
Added NQueens in Java

### PR DESCRIPTION
**Pull request by YASH01009**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)

### Description

(NQueens added in Java)

### Checklist

- [Y] I've read the contribution guidelines.
- [Y] I've checked the issue list before deciding what to submit.
- [Y] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Issue #1524 is solved here)

### CODE REVIEW

File: Java/cp/NQueens.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                    board[row][col] = false;
+                }
+        }
+    }
+
+  public static void main(String[] args) throws Exception {
+    Scanner scn = new Scanner(System.in);
+    int n = scn.nextInt();
+    boolean[][] board = new boolean[n][n];
+    //write your code here
```
Comment (HarshCasper) [inline Java/cp/NQueens.java]: This Comment is redundant. If you are solving a CP Problem from an Online Portal, prefer to add the link as well.

File: Java/cp/NQueens.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    
+  public static void solve(boolean[][] board, boolean[] cols, 
+    boolean[] ndiag, boolean[] rdiag, int row, String asf) {
+        
+        if(row == board.length) {
+            System.out.println(asf+'.');
+            return;
+        }
+        
+        for(int col=0; col<board.length; col++) {
```
Comment (HarshCasper) [inline Java/cp/NQueens.java:21]: Can you put apt comments on your Code for better explainability?

File: Java/cp/NQueens.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
+/**
+N-Queens problem is a famous problem
```
Comment (HarshCasper) [inline Java/cp/NQueens.java:2]: If possible, please add further information about the Algorithm

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Rest Code looks good to me! 

Please make these changes, and it will be good to merge 👍

**Comment by YASH01009** [comment]
@HarshCasper changes done !!

File: Java/cp/NQueens.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+Input :
+4
+Output :
+0-1, 1-3, 2-0, 3-2, .
+0-2, 1-0, 2-3, 3-1, .
+
+Space Complexity : O(n^2)
+Time Complexity : upperbounded by O(n^n)
+
+**/
```
Comment (HarshCasper) [inline Java/cp/NQueens.java]: Add a new line at the EOF

File: Java/cp/NQueens.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        for(int col=0; col<board.length; col++) {
+            if(cols[col] == false && ndiag[row+col] == false && 
+                rdiag[row-col+board.length-1] == false) {
+                    // place the queen
+                    cols[col] = true;
+                    ndiag[row+col] = true;
+                    rdiag[row-col+board.length-1] = true;
+                    board[row][col] = true;
+                    solve(board, cols, ndiag, rdiag, row+1, asf+row+'-'+col+", ");
+                    // backtrack
```
Comment (HarshCasper) [inline Java/cp/NQueens.java:33]: Can you make the comments more understandable?

File: Java/cp/NQueens.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        
+        if(row == board.length) {
+            System.out.println(asf+'.');
+            return;
+        }
+        
+        for(int col=0; col<board.length; col++) {
+            if(cols[col] == false && ndiag[row+col] == false && 
+                rdiag[row-col+board.length-1] == false) {
+                    // place the queen
```
Comment (HarshCasper) [inline Java/cp/NQueens.java:24]: Can you make the comments more understandable?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by YASH01009** [comment]
@HarshCasper changes done !!

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**